File: php/features.php
<?php
declare(strict_types=1);

// TemporaryEmailApi2 SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class TemporaryEmailApi2Features
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new TemporaryEmailApi2BaseFeature();
            case "test":
                return new TemporaryEmailApi2TestFeature();
            default:
                return new TemporaryEmailApi2BaseFeature();
        }
    }
}

File: py/temporaryemailapi2_sdk/features.py
# TemporaryEmailApi2 SDK feature factory

from temporaryemailapi2_sdk.feature.base_feature import TemporaryEmailApi2BaseFeature
from temporaryemailapi2_sdk.feature.test_feature import TemporaryEmailApi2TestFeature


def _make_feature(name):
    features = {
        "base": lambda: TemporaryEmailApi2BaseFeature(),
        "test": lambda: TemporaryEmailApi2TestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
